### PR TITLE
feat: adds azure metadata provider to cloudmeta pkg.

### PR DESCRIPTION
**Pull request by VAveryanov8**
This adds azure metadata provider to cloudmeta pkg.

https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=linux#usage 

I've tested this pr with official mock api from Azure - https://mockoon.com/mock-samples/azurecom-imds/ , but it's really far from real implementation, so it's better to validate this implementation against real Azure env. 

Fixes: #4129

---

Please make sure that:
- Code is split to commits that address a single change
- Commit messages are informative
- Commit titles have module prefix
- Commit titles have issue nr. suffix

### CODE REVIEW

**Comment by VAveryanov8** [comment]
This pr depends on https://github.com/scylladb/scylla-manager/pull/4154

File: pkg/cloudmeta/azure.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+	}
+}
+
+func defaultClient() *http.Client {
+	transport := http.DefaultTransport.(*http.Transport).Clone()
+	// we must not use proxy for the metadata requests - see https://learn.microsoft.com/en-us/azure/virtual-machines/instance-metadata-service?tabs=linux#proxies.
+	transport.Proxy = nil
+	return &http.Client{
+		// setting some generous timeout, it can be overwritten by using context.WithTimeout.
+		Timeout:   10 * time.Second,
```
[the remark this answers is not in the document]
Comment (mykaul) [reply pkg/cloudmeta/azure.go:38]: EC2 example - https://github.com/scylladb/scylladb/pull/11688

**Comment by karol-kokoszka** [comment]
> I've tested this pr with official mock api from Azure - https://mockoon.com/mock-samples/azurecom-imds/ , but it's really far from real implementation, so it's better to validate this implementation against real Azure env.

Create the request for IT to get the access to Azure.

> Dear Karol Kokoszka,
>
> We would like to let you know that we have received your request and a ticket has been created.
A support representative will be reviewing your request and will send you a personal response.
>
> To view the status of the ticket or add comments, please visit
https://scylladb.freshservice.com/helpdesk/tickets/441
>
> Thank you for your patience.
>
> Sincerely,
ScyllaDB IT Department

File: pkg/cloudmeta/azure.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (karol-kokoszka) [reply pkg/cloudmeta/azure.go:38]: @VAveryanov8 You can explore github.com/hashicorp/go-retryablehttp , the default http.Client doesn't provide retries unless you implement it on your own with custom Transport.

[the remark this answers is not in the document]
Comment (VAveryanov8) [reply pkg/cloudmeta/azure.go:38]: thanks, that make sense! 
I'll see if I can use some already imported package for that, I think I saw we had `backoff` package, it should provide retry functionality.

[the remark this answers is not in the document]
Comment (VAveryanov8) [reply pkg/cloudmeta/azure.go]: @karol-kokoszka thanks for the package suggestion - I decided to go with, it makes configuring retries very elegant.
PR is already updated!

**Comment by Michal-Leszczynski** [comment]
@VAveryanov8 please rebase on master so that it's easier to analyze the changes.

**Comment by VAveryanov8** [comment]
> @VAveryanov8 please rebase on master so that it's easier to analyze the changes. 

Done!

**Review by karol-kokoszka — APPROVED** [review]

File: pkg/cloudmeta/azure.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (VAveryanov8) [reply pkg/cloudmeta/azure.go]: @mykaul thanks for the review! Can we close this discussion?

[the remark this answers is not in the document]
Comment (karol-kokoszka) [reply pkg/cloudmeta/azure.go]: Just some context - all discussions/comments must be resolved to merge the PR in scylla-manger repo.

**Comment by VAveryanov8** [comment]
update: git rebase master

**Pull request merged**